### ISSUE TITLE
Not able to send from java email: (  400 Bad Request )  

### ISSUE THREAD

**Issue by aayoubb** [mailjet/mailjet-apiv3-java#134]
Hi 
I try to send an email from our java project but every time I have a code error 400. So I take the same request printed in the java console and send it from Postman with the same other elements then it works from Postman and i receive the email but from java I have always the code error 400 bad request and when i turn on the Mailjet debug mode the content is null.

Here the final JSON made by Java and inject from Postman 

```
{
	"Messages": [
		{
			"Variables": {
				"objet": " objet",
				"body": " body "
			},
			"From": {
				"Email": "no-reply@gmail.com"
			},
			"To": [
				{
					"Email": "send@gmail.com",
					"Name": ""
				}
			],
			"TemplateID": 111111,
			"TemplateLanguage": true,
			"Subject": "[[data:objet:\" anything\"]]"
		}
	]
}
```
The java code : 

```
/**
	 *  Send a message
	 * @return MailjetResponse
	 * @throws MailjetException
	 * @throws MailjetSocketTimeoutException
	 */
	public MailjetRequest sendEmailCore()  {
		return new MailjetRequest(Emailv31.resource)
				.property(Emailv31.MESSAGES, new JSONArray()
						.put(new JSONObject()
								.put(Emailv31.Message.FROM, new JSONObject()
									.put("Email", senderMail)
									.put("Name", senderName))
								.put(Emailv31.Message.TO, getRecipientsJson())
								.put(Emailv31.Message.TEMPLATEID, TEMPLATE_ID)
								.put(Emailv31.Message.TEMPLATELANGUAGE, true)
								.put(Emailv31.Message.SUBJECT, "[[data:objet:\" anything\"]]")
								.put(Emailv31.Message.VARIABLES, new JSONObject()
												.put("objet", " objet")
        										       .put("mailTitle", mailTitle)
												.put("body", body))));
	}

	private JSONArray getRecipientsJson(){
		JSONArray json = new JSONArray();
		for (String recipient : recipients) {
			json.put(new JSONObject()
						.put("Email", recipient)
						.put("Name", ""));
		}
		return json;
	}

	public MailjetResponse sendLoggedMail() throws Exception {
		MailjetClient client = new MailjetClient(API_KEY_PUBLIC, API_KEY_PRIVATE, new ClientOptions("v3.1"));
		MailjetRequest request = sendEmailCore();
		MailjetResponse response = null;
		try {
			client.setDebug(MailjetClient.VERBOSE_DEBUG);
			log.info(request.getBody());
			response = client.post(request);
		}  catch (final MailjetException | MailjetSocketTimeoutException e) {
		
			log.error("sendLoggedMail", e);
		} 
		return response;
	}
```
The MailJet log : 
=== HTTP Request ===
POST https://api.mailjet.com/v3.1/send
Accept-Charset:UTF-8
Accept:application/json
user-agent:mailjet-apiv3-java/v4.5.0
Content-Type:application/json
=== HTTP Response ===
Receive url: https://api.mailjet.com/v3.1/send
Status: 400
date:Tue, 01 Sep 2020 08:59:55 GMT
null:HTTP/1.1 400 Bad Request
content-length:177
x-mj-request-guid:0bb2a0ac-849d-4d80-82ed-a10a792e8d19
content-type:application/json; charset=UTF-8
Content:
null

Thank you for your help.

**Comment by ygModesto** [comment]
I have the same problem. The code works for me with version 4.2.1 and 4.4.0 but in 4.5.0 it returns the same error

**Comment by apenski** [comment]
same here. I looked into this a little bit and found out that `com.turbomanage.httpclient.AbstractHttpClient#execute`  gets an error/exception saying this: "No subject alternative DNS name matching api.mailjet.com found."

Same code works with 4.4.0

**Comment by a-vorniere** [comment]
Same here, back to 4.4.0.

**Comment by rethab** [comment]
I suspect the problem was introduced in https://github.com/mailjet/mailjet-apiv3-java/pull/128 where a default timeout of 10ms was added.

I managed to work around the issue by setting the timeout back to 8s, as it was before (default by upstream library, see https://github.com/turbomanage/basic-http-client/blob/master/http-client-java/src/main/java/com/turbomanage/httpclient/AbstractHttpClient.java#L43).

The timeout can be specified using the `ClientOptions` which are passed to the `MailjetClient`'s constructor.

@cweidenkeller perhaps the default should be a bit higher than 10ms?
